base/stl_helpers.hpp

```diff
@@ -45,12 +45,18 @@ struct Comparer<false, T, C>
 
 // Sorts and removes duplicate entries from |v|.
 template <typename T>
-void SortUnique(std::vector<T> & v)
+void SortUnique(vector<T> & v)
 {
   sort(v.begin(), v.end());
   v.erase(unique(v.begin(), v.end()), v.end());
 }
 
+template <typename T, class TFn>
+void EraseIf(vector<T> & v, TFn fn)
```

Comment:
Так один раз пойдет ) Пользователям, в основном, будет пофиг, будут лямбды пихать.

> On Mar 23, 2016, at 23:47, Viktor Govako <notifications@github.com> wrote:
> 
> In base/stl_helpers.hpp <https://github.com/mapsme/omim/pull/2454#discussion_r57234216>:
> 
> >  {
> >    sort(v.begin(), v.end());
> >    v.erase(unique(v.begin(), v.end()), v.end());
> >  }
> >  
> > +template <typename T, class TFn>
> > +void EraseIf(vector<T> & v, TFn fn)
> Еще раз. Допусти я сделаю &&. Все равно внутри в remove_if он пойдет по копированию.
> Не нужно вводить пользователя в заблуждение.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2454/files/de3aa465c10122ec1a980ba6162ad33dac4dd2ea#r57234216>